_components/CodeSamples.tsx

```diff
@@ -0,0 +1,329 @@
+// {% extends "base.html.jinja" %}
+import React, { useState } from "react";
+import preval from "babel-plugin-preval/macro";
+import { usePathPrefix } from "@redocly/developer-portal/ui";
+import DropdownButton from "react-bootstrap/DropdownButton";
+import Dropdown from "react-bootstrap/Dropdown";
+// {% block bodyclassNamees %}no-sidebar{% endblock %}
+// {% block mainclassNamees %}landing page-community{% endblock %}
+
+// {% block main %}
+
+interface CodeSample {
+  href: string;
+  title: string;
+  description: string;
+  langs: Array<string>;
+}
+// TODO: need to be updated once repo is migrated
+const githubLink = "https://github.com/amarantha-k/xrpl-org-dev-portal";
+
+export default function CodeSamples() {
+  const prefix = usePathPrefix();
+  let [langSelected, setLangSelected] = useState("All");
+  // scanner script to go through all code-samples
+  const codeSamples = preval`const fs = require("fs");
```

Comment:
Can you just do this directly as javascript code instead of using `preval`?

Reply:
React can't execute fs operations at runtime, this is a workaround